### DISCUSSION TITLE
Why does supabase.auth.signIn() resolve before the user even logs in?

### DISCUSSION THREAD

**Discussion by Squirrelcoding** [supabase/supabase#6376] [Questions]
This is probably a dumb question but bear with me.

I am trying to have code execute after the snippet below finishes
```js
    await supabase.auth.signIn({
        provider: "google"
    });
```
I _need_ to have a session by the time it finishes, as other code depends on it. But the thing is, the function resolves even before the user logs in! This leads to `Session` being `null` and the code depending on it doesn't work as expected. I tried doing the following:
```js
    supabase.auth.signIn({
        provider: "google"
    }).then((returnedVal) => {
        // Other code goes here
    })
```

But that doesn't work either because the promise resolves when the user is selecting their google account, and naturally, the session wouldn't be defined yet. So does anybody know of a way to have code execute once the user selects the google account and the session exists?

**Comment by GaryAustin1** [comment]
For third party logins, the return call comes back to your default page unless you set the redirectTo parameter to come back to a different route.  If you have the  onAuthStateChange handler setup it will generate a SIGNED_IN event you can then do your signin process work on.  https://supabase.com/docs/reference/javascript/auth-onauthstatechange
The signIn function basically just returns null because you are going to launch a new page when signIn is done with the 3rd party.
SignIn with email/password does return because it is just an auth/database call to confirm.
